📝 Rename Websocket by WebSocket

Kludex opened the pull request:
This is fundamentally unnecessary, so the rejecting this PR is understandable. But the correct name is WebSocket.

andrewgodwin commented:
I don't object to it, but if you want to land this you'll need to do it in a backwards-compatible fashion (so the old names work but raise a warning)

Kludex commented:
I've used [PEP 562](https://www.python.org/dev/peps/pep-0562/) idea here, but as it was introduced in 3.7, I've used the [backport implementation for 3.6](https://github.com/facelessuser/pep562).

I've also added tests to check that the warning is being raised on the wanted types.

andrewgodwin commented:
I'd rather just do it the "old fashioned" way if we could - expose them as actual top-level module objects rather than bringing in extra code just for this.

Kludex commented:
I'm not following you. :thinking: How am I supposed to warn the user about the deprecation at import time without this implementation?

Or you just meant to not use `__all__` and `__deprecated__`?

andrewgodwin commented:
Oh, I forgot they were type classes for a second, sorry. Thought they were actual classes where you could do it with a mixin on instantiation.

In that case, let's vendor pep562 in provided the license matches - I don't want more dependencies!

Kludex commented on asgiref/_pep562.py:
```diff
@@ -0,0 +1,59 @@
+        self._get_dir: Optional[Callable[..., List[str]]] = getattr(
+            self._module, "__dir__", None
+        )
+        sys.modules[name] = self  # type: ignore[assignment]
```
`sys.modules` expects a `Module` and `self` is a `Pep562`, jfyk.

Kludex commented:
There's a mismatch between the linters, `isort` and `black` are reformatting themselves on the `pre-commit` :thinking:

Kludex commented:
Notes about the changes: 
- I've added `pep562` as private module, so we can remove easily by the [3.6 EOL](https://endoflife.date/python).
- I've removed all the unnecessary code on `_pep562.py` (and tried to modify the less I could, jfyk).

andrewgodwin commented:
I think sticking with `black`'s default length of 88 is the best approach - let's handle that in a separate PR. I'll merge this in now so we can avoid mixing linting in here - I think it looks good now, so thanks for bearing with my requests!

Kludex commented:
Thank you! :tada:

Kludex commented:
This is not `mypy` compliant. A project with `asgiref` not pinned, will receive this on the upgrade:
```python
+ mypy --show-error-codes
uvicorn/middleware/proxy_headers.py:13: error: Module "asgiref.typing" has no attribute "WebsocketScope"; maybe "WebSocketScope"?  [attr-defined]
```

Which I think it's expected. but I'm mentioning because I'm not sure you were aware. @andrewgodwin
